### PR TITLE
19 2.1 - Adding VRC CPU minning Coin (ask from bumbu100 on BTCTALK)

### PR DESCRIPTION
**Pull request by cartesien**
Adding/replacing Vericoin coin (and VeRiumMiner) for CPU mining ONLY

More explanations about ASIC and GPU resistant Vericoin coin here :
Steemit : https://steemit.com/cryptocurrency/@jonipilo/verium-and-vericoin-the-most-profitabale-altcoin-to-mine
BTCTALK thread : https://bitcointalk.org/index.php?topic=1540023.0

I'm skeptical because VRC coin is scrypt algorithm so i don't understand why it's not mineable with ASIC L3+ miner (but it seems to use scrypt² not really scrypt). This miner was asked in the tthread (by bumbu100), so i propose some code to mine it on nvOC.

I think papampi and leenoox or damnmad can propose more complex conditionnal loop for testing if you choose XMR or VRC for CPU minning but i'm not sure if it's really a good choice or idea. So i let them to decide if we have to implement a conditionnel loop for this.

Note : What I propose to modify here induces that you will not be able to mine monero with the CPU, but you can always do it with the GPU  by the other program xmr-stak-Gpu. You must be aware of this implication before you start modifying the 3main script.

1/ Download and unzip dedicated cpuminer fork : https://github.com/fireworm71/veriumMiner/releases/download/v1.4/cpuminer_1.4_linux_x64_O2_GCC7.zip
or
compile it from source : https://github.com/fireworm71/veriumMiner
(remmember ./autogen.sh; ./configure.sh; ./build.sh to compile the miner for your own system)

2/ Place it in /home/m1/VeRiumMiner (so, you have to move it and rename his directory in /home/m1 to have the same path like in the code below)

3/ Find plusCPU section in 3main script bash file (line 290 for me in nvOC 19-2.0):

Code:

if [ $plusCPU == "YES" ] && [ $AUTO_START_MINER == "YES" ]
then
  HCD='/home/m1/cpuOPT/cpuminer'
  XMRADDR="$XMR_ADDRESS.$XMR_WORKER"
  echo ""
  echo ""
  echo "LAUNCHING:  plusCPU"
  if [[ `ps -ef |grep cpuminer |grep -v grep |wc -l` -eq 0 ]]
  then
    if [ $LOCALorREMOTE == "LOCAL" ]
    then
      guake -n $HCD -r plusCPU -e "$HCD -a cryptonight -o stratum+tcp://$XMR_POOL:$XMR_PORT -u $XMRADDR -p x -t $threadCOUNT"
    else
      screen -dmS plusCPU $HCD -a cryptonight -o stratum+tcp://$XMR_POOL:$XMR_PORT -u $XMRADDR -p x -t $threadCOUNT
    fi
    echo ""
    echo "plusCPU process in guake terminal Tab (f12)"
    echo ""
    running=""
  fi

 
  4/ replace code above with :
  
 
Code:

 if [ $plusCPU == "YES" ] && [ $AUTO_START_MINER == "YES" ]
then
  HCD='/home/m1/VeRiumMiner/cpuminer'
  VRCADDR="$VRC_ADDRESS.$VRC_WORKER"
  echo ""
  echo ""
  echo "LAUNCHING:  plusCPU"
  if [[ `ps -ef |grep cpuminer |grep -v grep |wc -l` -eq 0 ]]
  then
    if [ $LOCALorREMOTE == "LOCAL" ]
    then
      guake -n $HCD -r plusCPU -e "$HCD -o stratum+tcp://$VRC_POOL:$VRC_PORT -u $VRCADDR -p $VRC_PWD -t $threadCOUNT"
    else
      screen -dmS plusCPU $HCD -a cryptonight -o stratum+tcp://$VRC_POOL:$VRC_PORT -u $VRCADDR -p $VRC_PWD -t $threadCOUNT
    fi
    echo ""
    echo "plusCPU process in guake terminal Tab (f12)"
    echo ""
    running=""
  fi

 
 5/ Add this in 1bash at the end of indivuduals coins :

Code:

# VERICOIN (VRC) 
  # ## No Need to add add stratum+tcp:// to server address
VRC_WORKER="$WORKERNAME"						# WorkerName  : remember that you have to create it in your account pool page !
VRC_ADDRESS="YourWebloginPool"						# Local Wallet Address
VRC_POOL="vrm.n3rd3d.com" 						# 1% pool fee
VRC_PORT="3332"								# It depnds on your CPU Core (see the pool page to use the best port for you)
VRC_PWD="x"

VeRiumMiner use all thread avalilable on your CPU by default but you can specify how much threads at the begenning of 1bash file with this variable between plusCPU and CONTROL_GPU_LED :

Code:

threadCOUNT="1"             # Number of CPU threads to use for plusCPU mining. Please check how many threads your CPU has.

Not tested on my rig but i test minning with typing manually the commands and it works on nvOC. I compile myself but i think pre-compiled file is ok.
It should be verified that the cpuminer that is already installed in nvOC does not allow to directly undermine this crypto currency. I did not do it either.

Happy minning Vericoin !

### CODE REVIEW

**Comment by cartesien** [comment]
1/ I just update my miners by using your script update of the dédicated thread on btctalk thread.
https://bitcointalk.org/index.php?topic=1854250.msg29101637#msg29101637
I had some problems. I explain myself.
I propose some modifications in directory miners arboresence to simplify updating miners manually for all (including Linux newbies) : 
I made theses modifications myself on my nvOC v19-2.0 and i'm not great experienced like you or stubo or leenoox.

I think it's more easy for anyone to copy or replace in a directory named [latest] for all miners where there is only miners binaries files (and their necessarily librairies), than to make a new directory for each version and modify 0miner and 3main file and ...etc But you can make some little complex scripts that permit to keep old miners if necesary
So i propose to make one directoty for [olds] versions, and one directory for [latests].

Like this 
/home/m1/miners/olds/{directory name of miner}/version/ 
for old versions of miners, and
like this for latest versions of miners : 
/home/m1/miners/latest/{directory name of miner}/*

More details of what i propose on this image : https://imgur.com/Vx36BhM

2/ It's not the first time, I had to compile these three miners this evenning : 
- KTccminer
- NAccminer (with bignum error...)
- TPccminer (i'm not sur about bignum because i don't hand over ssl librarie in his place yet)

In addition, i read a lot of time that it's preferable to compile on your own machine than to download from anyone who compile it on his own too.

So i propose you add compilations of theses minning softwares by default in your miners update scripts.
What do you think about this proposition ?

3/ For updating miners softwares, i think you should make a master script that calls each individuals scripts for each miners.

4/ I modify Pasc dierctory because when  i test miner, i see it's nicehash sgminer version. So i called it sgminer-nicehash

To summarize : 
1/ Big modifications of directories for miners (30 min)
2/ Adding compilation in update script for theses 3 miners i quote above.
3/ Master update script calls each miner individual update script
4/ Renaming pasc folder in his real name.


Continue your good job to make nvOC a real good alternative for mining than ETHos which claim money or fees. 

I knew you were going to respond to those who want to "pay" for your good but heavy and long scripting work.

I do not understand why they ask elsewhere, they seem not to see your crypto address in the scripts ... If they want they can give on these addresses. I imagine many others give without saying anything and that's fine. Finally, I hope for you ...
